routing/mock_graph_test.go

```diff
@@ -131,16 +130,25 @@ func (m *mockGraph) addNode(node *mockNode) {
 // Ignore linter error because addChannel isn't yet called with different
 // capacities.
 // nolint:unparam
-func (m *mockGraph) addChannel(node1, node2 *mockNode, capacity btcutil.Amount) {
-	id := m.nextChanID
-	m.nextChanID++
+func (m *mockGraph) addChannel(id uint64, node1id, node2id byte,
+	capacity btcutil.Amount) {
+
+	node1pubkey := createPubkey(node1id)
+	node2pubkey := createPubkey(node2id)
+
+	if _, exists := m.nodes[node1pubkey].channels[node2pubkey]; exists {
+		m.t.Fatal("channel already exists")
```

Comment:
In the current test setup, multiple channels between nodes are not recognized. but this can also be simulated by modifying the balance update logic of a (fake) node.

Reply:
In general, I am still developing a complete plan on where the integrated routing test concept should develop towards. Where to exactly make the cut and mock.

Reply:
Ah, okay for now then.